core/txpool/blobpool/blobpool.go

```diff
@@ -0,0 +1,1649 @@
+//     either, the engine_newPayload needs to give them to us, and we cache them
+//     until finality to support reorgs without tx losses.
+//
+// Whilst some constraints above might sound overly aggressive, the general idea is
+// that the blob pool should work robustly for its intended use case and whilst
+// anyone is free to use blob transactions for arbitrary non-rollup use cases,
+// they should not be allowed to run amok the network.
+//
+// Implementation wise there are a few interesting design choices:
+//
+//   - Adding a transaction to the pool blocks until persisted to disk. This is
+//     viable because TPS is low (2-4 blobs per block initially, maybe 8-16 at
+//     peak), so natural churn is a couple MB per block. Replacements doing O(n)
+//     updates are forbidden and transaction propagation is pull based (i.e. no
+//     pileup of pending data).
+//
+//   - When transactions are chosen for inclusion, the primary criteria is the
+//     signer tip (and having a basefee/data fee high enough of course). However,
+//     same-tip transactions will be split by their basefee/datafee, preferring
+//     those that are closer to the current network limits. The idea being that
+//     very relaxed ones can be included even if the fees go up, when the closer
+//     ones could already be invalid.
+//
+// When the pool eventually reaches saturation, some old transactions - that may
+// never execute - will need to be evicted in favor of newer ones. The eviction
+// strategy is quite complex:
+//
+//   - Exceeding capacity evicts the highest-nonce of the account with the lowest
+//     paying blob transaction anywhere in the pooled nonce-sequence, as that tx
+//     would be executed the furthest in the future and is thus blocking anything
+//     after it. The smallest is deliberately not evicted to avoid a nonce-gap.
+//
+//   - Analogously, if the pool is full, the consideration price of a new tx for
+//     evicting an old one is the smallest price in the entire nonce-sequence of
+//     the account. This avoids malicious users DoSing the pool with seemingly
+//     high paying transactions hidden behind a low-paying blocked one.
+//
+//   - Since blob transactions have 3 price parameters: execution tip, execution
+//     fee cap and data fee cap, there's no singular parameter to create a total
+//     price ordering on. What's more, since the base fee and blob fee can move
+//     independently of one another, there's no pre-defined way to combine them
+//     into a stable order either. This leads to a multi-dimensional problem to
+//     solve after every block.
+//
+//   - The first observation is that comparing 1559 base fees or 4844 blob fees
+//     needs to happen in the context of their dynamism. Since these fees jump
+//     up or down in ~1.125 multipliers (at max) across blocks, comparing fees
+//     in two transactions should be based on log1.125(fee) to eliminate noise.
```

Comment:
Is this true that they will increase at max of 1.125 of the current base fee? I think that may be incorrect for EIP-1559

Reply:
cc @abi87 who just pointed this out to me today

Reply:
It seems to me it was the original intention (see https://github.com/ethereum/EIPs/issues/1559) but that is not strictly true anymore in https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md nor in the implementation.
To be clear, I understand fee changes are still capped, but not of 1/changeDemon. changeDemon seems a smoothing parameter to control fee dynamics.